astroid/nodes/node_classes.py

```diff
@@ -2020,7 +2023,7 @@ def postinit(self, nodes: typing.List[NodeNG]) -> None:
         """
         self.nodes = nodes
 
-    def scope(self):
+    def scope(self) -> Optional["LocalsDictNodeNG"]:
```

Comment:
The function body here isn't actually safe either. It should be
```py
if self.parent and self.parent.parent:
    return self.parent.parent.scope()
return None
```

Reply:
After discussion above, this is no longer relevant.